core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -550,6 +575,7 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeString(cursor.key);
             cursor.value.writeTo(out);
         }
+        out.writeStringArray(primaryCandidateAllocationIds.toArray(new String[primaryCandidateAllocationIds.size()]));
```

Comment:
I am not sure if the iteration order is important somewhere down the road but if we serialize over the network and build another set from it we can end up with different iteration order. If it's not important that's fine just wanna raise it.